pandas/tests/extension/base/reduce.py

```diff
@@ -8,6 +9,46 @@
 from .base import BaseExtensionTests
 
 
+def generate_error_message(dtype, op_name):
```

Comment:
This is too complicated - [don't put logic in tests](https://testing.googleblog.com/2014/07/testing-on-toilet-dont-put-logic-in.html), each tests can have its own message

Reply:
Though actually, having looked at this more closely, I don't think there's an easy way to do this, what you've done is fine